### PR TITLE
fix(chart): Skip README when it cannot be found

### LINKED ISSUE #3226: Redirect to DeploymentForm when chart is missing README
Hi guys, me again :)

### Proposal
When there are  helm charts w/out a README users a greeted w/ a blank page that doesnt seem to provide much value (just some 'no README' text and the version controller dropdown). 
image
I was wondering how you guys would feel about just doing a direct link to the deployment form when there is no README for the associated chart.

As of right now I don't see any real drawbacks from making this the default behavior but feedback is greatly appreciated. As per all of my issues/feat requests, Im willing to take ownership of this feature.

### PR DESCRIPTION
**Pull request by aanthonyrizzo**
Signed-off-by: Anthony Rizzo <36711320+aanthonyrizzo@users.noreply.github.com>

### Description of the change

This PR adds a logical statement to redirect and not render the `<ChartReadme />` when there is a README error. Since I only want to redirect when the file is not found, I check the error as such `.includes("not found")`. I cant see why having this as the default behavior would cause issues but this can easily be controlled w/ a config variable if we wanted.

### Benefits

This change will circumvent the no README error and direct the user to the deployment form

### Possible drawbacks

I don't see any but I could be wrong

### Applicable issues

Fixes #3226

### CODE REVIEW

**Comment by absoludity** [comment]
See comment at https://github.com/kubeapps/kubeapps/issues/3226#issuecomment-899878636 for more info about this PR.

**Review by antgamdia — CHANGES_REQUESTED** [review]
_(just updating the status of the PR according to what it was said)_

**Comment by aanthonyrizzo** [comment]
Sorry guys I haven't had time to work on this recently. I will get back on this next week.

**Comment by ppbaena** [comment]
Thank you @aanthonyrizzo for the info! 
We have configured this bot mainly to stale discontinued issues, but also as a kindly reminder for situations like you mention. Next week more!

File: dashboard/src/containers/RoutesContainer/Routes.tsx [hunk 1]
```diff
@@ -29,8 +29,12 @@ const privateRoutes = {
   "/c/:cluster/ns/:namespace/apps/:pluginName/:pluginVersion/:releaseName/upgrade": AppUpgrade,
   "/c/:cluster/ns/:namespace/apps/new/:repo/:pluginName/:pluginVersion/:id/versions/:version":
     DeploymentForm,
+  "/c/:cluster/ns/:namespace/apps/new/:repo/:pluginName/:pluginVersion/:id/versions/":
```
[the remark this answers is not in the document]
Comment (antgamdia) [reply dashboard/src/containers/RoutesContainer/Routes.tsx:32]: Mmmm, `latest` would imply a single version... but you are still able to select multiple ones from the dropdown... so, perhaps `versions` fits better here, no? No strong opinion anyway,

File: dashboard/src/components/DeploymentForm/DeploymentForm.tsx [hunk 2]
```diff
@@ -132,6 +132,7 @@ export default function DeploymentForm() {
   };
 
   const selectVersion = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    console.log(e.target.value);
```
Comment (antgamdia) [inline dashboard/src/components/DeploymentForm/DeploymentForm.tsx:135]: leftover :P

**Review by antgamdia — COMMENTED** [review]
Let us know when it is ready for review so that we can pull and test the changes. And, as usual, thanks a lot for your contributions!

File: dashboard/src/components/DeploymentForm/DeploymentForm.tsx (hunk 2 above)
Comment (aanthonyrizzo) [reply dashboard/src/components/DeploymentForm/DeploymentForm.tsx:135]: 🤦‍♂️

**Comment by aanthonyrizzo** [comment]
> Sorry for the delay on this one Anthony. If you can rebase with current master, I'll try to get it reviewed and landed ASAP, early in our next cycle.

Hi I also want to apologize, crazy week, I should be able to finish this early next week (Monday has the potential to be very busy so it might be Tuesday)

**Comment by antgamdia** [comment]
Oops, it seems our bot closed it. @aanthonyrizzo, I guess you didn't have a chance to get back to it, don't worry; do you want us to cherry-pick your changes in another PR?

**Comment by aanthonyrizzo** [comment]
> Oops, it seems our bot closed it. @aanthonyrizzo, I guess you didn't have a chance to get back to it, don't worry; do you want us to cherry-pick your changes in another PR?

Yeah I was going to open another PR since this got so far behind with all of the route changes you guys made when changing the chart => package nomenclature but I've been so swamped w/ other work I just havent had the time.  If you could that would be super awesome :)

**Comment by antgamdia** [comment]
Great, awesome! I just didn't want to jump in w/o letting you know first. No problem, sometimes we all got overwhelmed with tons of things on our plate hehe.
Thanks for your words, glad you liked it!!   :blush:

**Pull request closed without merging**